### PR TITLE
Remove support for TPRs and K8s 1.6

### LINKED ISSUE #1421: Drop support for Kubernetes 1.6.x
Rook officially maintains support for 1.6.x and a YMMV support for 1.5.x. I think it is time to drop support for both of them. Rook strives to run natively and to be deeply integrated on Kubernetes. Rook does this by leveraging the K8S extensibility via CRDs.

CRDs were introduced on 1.7. Prior to that, Rook relied on TPRs which has been deprecated and removed from Kubernetes.

Dropping support for K8S 1.6x will make the upgrade and support a lot smoother in the future. We dont have to worry about converting all our TPR objects into CRDs.

### PR DESCRIPTION
**Pull request by travisn**
Signed-off-by: travisn <tnielsen@redhat.com>

Remove support for K8s 1.6, including TPRs. Rook v0.8 must be installed on K8s 1.7+. Installing on an older version of Kubernetes can, however, still be done with Rook v0.7.

Resolves #1421

Checklist:
- [X] Documentation has been updated, if necessary.
- [X] Pending release notes updated with breaking and/or notable changes, if necessary.

### CODE REVIEW

File: PendingReleaseNotes.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 - Rook-Operator no longer creates the resources CRD's or TPR's at the runtime. Instead, those resources are provisioned during deployment via `helm` or `kubectl`.
 
 ## Breaking Changes
+- Removed support K8s 1.6, including the legacy TPRs.
```
Comment (jbw976) [inline PendingReleaseNotes.md]: I would use less acronyms in this message:
> Removed support for Kubernetes 1.6 and older, including the legacy Third Party Resources (TPRs).  Rook now supports only Kubernetes 1.7 and newer releases.

File: pkg/daemon/agent/flexvolume/attachment/crd.go [hunk 2]
```diff
@@ -51,22 +43,7 @@ type crd struct {
 
 // CreateController creates a new controller for volume attachment
 func New(context *clusterd.Context) (Attachment, error) {
-
-	// CRD is available on v1.7.0. TPR became deprecated on v1.7.0
```
Comment (jbw976) [inline pkg/daemon/agent/flexvolume/attachment/crd.go]: What's the experience for someone deploying on 1.6 or older? do they get a decent error message or is it somehow obvious what the problem is?

File: tests/integration/base_file_test.go [hunk 3]
```diff
@@ -60,7 +61,7 @@ func runFileE2ETest(helper *clients.TestClient, k8sh *utils.K8sHelper, s suite.S
 	logger.Infof("Step 4: Read from file system")
 	read, rdErr := helper.FSClient.Read(filePodName, fileMountPath, "fsFile1", namespace)
 	require.Nil(s.T(), rdErr)
-	require.Contains(s.T(), read, "Smoke Test Data for file system storage", "make sure content of the files is unchanged")
+	assert.Contains(s.T(), read, "Smoke Test Data for file system storage", "make sure content of the files is unchanged")
```
Comment (jbw976) [inline tests/integration/base_file_test.go]: why the change from require to assert?

File: pkg/daemon/agent/flexvolume/attachment/crd.go (hunk 2 above)
Comment (travisn) [reply pkg/daemon/agent/flexvolume/attachment/crd.go]: I would expect that when you create the rook-operator.yaml, it will immediately complain that it can't create the CRDs.

File: tests/integration/base_file_test.go (hunk 3 above)
Comment (travisn) [reply tests/integration/base_file_test.go]: this is really a non-fatal error and should not block the test from continuing. It masked other failures that I could have fixed at the same time.

File: Documentation/cluster-crd.md [hunk 4]
```diff
@@ -81,11 +81,11 @@ Below are the settings available, both at the cluster and individual node level,
...
-- `nodeAffinity`: kubernetes [NodeAffinity](https://kubernetes.io/docs/api-reference/v1.6/#nodeaffinity-v1-core)
-- `podAffinity`: kubernetes [PodAffinity](https://kubernetes.io/docs/api-reference/v1.6/#podaffinity-v1-core)
-- `podAntiAffinity`: kubernetes [PodAntiAffinity](https://kubernetes.io/docs/api-reference/v1.6/#podantiaffinity-v1-core)
-- `tolerations`: list of kubernetes [Toleration](https://kubernetes.io/docs/api-reference/v1.6/#toleration-v1-core)
+A Placement configuration is specified (according to the kubernetes PodSpec) as:
+- `nodeAffinity`: kubernetes [NodeAffinity](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#node-affinity-beta-feature)
+- `podAffinity`: kubernetes [PodAffinity](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#inter-pod-affinity-and-anti-affinity-beta-feature)
+- `podAntiAffinity`: kubernetes [PodAntiAffinity](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#inter-pod-affinity-and-anti-affinity-beta-feature)
+- `tolerations`: list of kubernetes [Toleration](https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/)
```
Comment (galexrt) [inline Documentation/cluster-crd.md]: I think it is better to keep the api reference links but increase the version to `1.7` instead of replacing them with other docs.

File: PendingReleaseNotes.md [hunk 5]
```diff
@@ -8,6 +8,7 @@
 - Rook-Operator no longer creates the resources CRD's or TPR's at the runtime. Instead, those resources are provisioned during deployment via `helm` or `kubectl`.
 
 ## Breaking Changes
+- Removed support for Kubernetes 1.6, including the legacy Third Party Resources (TPRs).
```
Comment (galexrt) [inline PendingReleaseNotes.md]: I think changing `1.6` to `<= 1.6` (or something like that) is better as 1.5 support is also finally dropped.

File: Documentation/cluster-crd.md (hunk 4 above)
Comment (travisn) [reply Documentation/cluster-crd.md]: @galexrt what links would you suggest? The 1.6 and 1.7 documentation has a warning at the top that the docs are no longer maintained, and those links didn't seem to go to a specific topic anyway.

File: PendingReleaseNotes.md (hunk 5 above)
Comment (travisn) [reply PendingReleaseNotes.md]: sure, i'll make that more clear thanks
